python/cudf/cudf/core/series.py

```diff
@@ -6034,6 +6034,18 @@ def explode(self, ignore_index=False):
         3       5
         dtype: int64
         """
+        if is_struct_dtype(self._column.dtype):
+            cols = [key for key in self.dtype.fields]
+            results = []
+            for row in self.to_arrow():
+                row_results = [str(row[col]) for col in cols]
+                results.append(row_results)
+
+            out = cudf.DataFrame(results, columns=cols)
```

Comment:
What is the structure of the `out` you want here? The device to host transfer + nested loop may be quite slow

Reply:
the idea was that it would essentially look the same as the `struct.explode()` functionality that's already been implemented, so like: 
![image](https://user-images.githubusercontent.com/53962159/127228665-2e0761da-23df-4cd0-a3a4-e4e1e383fe46.png)

Reply:
You may be able to accomplish that without going to the CPU:

```python
import cudf
​
s = cudf.Series([
    {"a":5, "b":10},
    {"a":3, "b":7},
    {"a":-3, "b":11},
])
​
results = []
for key in s.dtype.fields:
    results.append(s.struct.field(key))
​
out = cudf.concat(results, axis=1)
out.columns = s.dtype.fields
print(out)
   a   b
0  5  10
1  3   7
2 -3   11
```

Separately, we may want to think more broadly about what we want the behavior to be for functionality that "kind of" exists in pandas. Pandas doesn't have struct columns, but does allow exploding of an object column containing dictionaries. However, the explode does not behave like exploding a struct column in Hive, Spark, etc. Instead, it behaves like exploding a list column (which it doesn't technically have either).

Because of that, we might need to special case an `explode` operator in dask-cuDF anyway, rather than rely on Dask to appropriately delegate to the cuDF `explode` from the existing one in Dask.DataFrame.

For now, I'd suggest we consider holding off on `series.explode()`, and building the `dask_series.struct.explode()` functionality once we land #8658

Reply:
cc @shwina @VibhuJawa (as they might disagree)

Reply:
I agree, I don't think we want the `Series.explode()` to support this just yet (unless Pandas does so). `dask_series.struct.explode()` can work off of `series.struct.explode()`.

Reply:
in Pandas, I get: 
`
import pandas as pd

s = pd.Series([{"a": 1, "b": "x"},
                {"a": 2, "b": "y"},
                {"a": 3, "b": "z"},
                {"a": 4, "b": "a"}])
`
`
s.explode()
0    a
0    b
1    a
1    b
2    a
2    b
3    a
3    b
dtype: object
`

Reply:
Thanks for the example 👍 .

This is the "traditional explode" mentioned above and what we do with lists. The "lateral explode" is particularly common for structs, as the field names often map to actual features

Reply:
sounds good - I'll close this PR then, as it has a different functionality than the desired `dask_series.struct.explode()`